py/torch_tensorrt/dynamo/conversion/impl/select.py

```diff
@@ -127,9 +148,7 @@ def index(
         )
         index = adv_indx_indices[0]
         _LOGGER.debug(f"The advanced index indices is {adv_indx_indices}")
-        gather_layer = ctx.net.add_gather(input, indices_tensor, index)
-        set_layer_name(gather_layer, target, name + "_index_gather", source_ir)
-        return gather_layer.get_output(0)
+        return gather(ctx, target, source_ir, name, input, index, indices_tensor)
```

Comment:
Line `126`/`147` needs to be renamed since it uses `name + f"_parameter_to_fp32_tensor"`, which also appears in the `gather` function. This could cause a duplicate name error in edge cases